hub/api/dataset.py

```diff
@@ -190,6 +183,16 @@ def __iter__(self):
         for i in range(len(self)):
             yield self[i]
 
+    def _load_meta(self):
+        if dataset_exists(self.storage):
+            self.meta = DatasetMeta.load(self.storage)
+            for tensor_name in self.meta.tensors:
+                self.tensors[tensor_name] = Tensor(tensor_name, self.storage)
+        elif len(self.storage) > 0:
+            raise PathNotEmptyException
```

Comment:
why? we may want to allow this

Reply:
Why should we allow writing a dataset to a non-empty folder?

Reply:
This is anyway not in the context of this PR, I only moved some existing stuff into a function

Reply:
ok cool no worries